### ISSUE TITLE
[Future Request] Sending software command for reboot to bootloadre in GUI

### ISSUE THREAD

**Issue by MattWestb** [thegroove/esphome-zbbridge#4]
thya ZBGW we was adding script for rebooting the NCP in bootloader mode.
If having one hardware reset and on force bootloader pin its better but some hardware dont have that implanted but the tuya is not having that.

By sending one reset command and then reading the EZSP version for knowing witch command set that need being used and then sending the right frame with reboot in bootloader mode.
The EZSP is using frame counters for all commands so need doing it in right sequence so not needing recalculating the frames all the time (static sequence).

The script is in this issue https://github.com/banksy-git/lidl-gateway-freedom/pull/5.
The commands is documented and if you like have more information i can digging for getting it.

Best is making it support protocol version 7 and 8 and older is not very likely finding on newer devices.

I finding implanting x-modem is not needed better the user is using one terminal program like minicom (Linux) or extra PuTTY (Windows) for doing the upload.

If getting one firmware that is possible using in router mode is also possible making on script for leaving and joining the network with serial commands that can  being triggered with the GUI.

**Comment by MattWestb** [comment]
Also one more thing.
I was helping Tasmota team testing the implanting the EZSP art (was fighting with the IKEA Zigbee 3 devices was not working then the trustcenter link security frame counter being retested every reboot) and also the tuya ZBGW.
Then having connected one coordinator over TOC/IP its can being god releasing the port and using bellows for sending commands to the coordinator.
That is not possible then ZHA is up and running so i changing  the port in ZHA and restarting HA and ZHA is not starting but can open on docker CLI and sending commands with bellows to the coordinator.

If it was possible changing the TCP/PI port "on the fly" (if possible without rebooting the ESP) it wold being great for doing firmware upgrade of the EFR32 chip or changing parameters in it that is not possible doing then ZHA is connected.

Perhaps one GUI "temp TCP/IP port XXXX" and toggling "activating/deactivating".

That shall being great for doing system maintenance and also temporary using the coordinator as sniffing Zigbee traffic in the network and so on.

**Comment by thegroove** [comment]
Yes that's smart, it wouldn't be hard to expose an "update mode" switch that closes the active serial server and opens it up on another port for external tools. I'll implement this approach when I spend the time on the firmware upload procedure.

**Comment by Hedda** [comment]
FYI, maybe relatest comment from xsp1989 posted on router firmware posted on https://github.com/arendst/Tasmota/discussions/11536

"This firmware can ground the PA00 5S into pairing mode, and ground the RST to restart routing. UART is used to transmit data transparently.
